docs/CIS2Contract.md

```diff
@@ -0,0 +1,246 @@
+
+See the signing a transaction section for the [common package](../packages/common/README.md#sign-an-account-transaction) for create an `AccountSigner`.
+
+### Using an `AccountSigner`.
+This relies on using private keys for the sender account to sign the transaction before submitting to the node.
+
+```js
+const tokenId = ''; // HEX string representing a token ID defined in the contract.
+const from = '4UC8o4m8AgTxt5VBFMdLwMCwwJQVJwjesNzW7RPXkACynrULmd';
+const to = '3ybJ66spZ2xdWF3avgxQb2meouYa7mpvMWNPmUnczU8FoF8cGB'; // An account receiver.
+// const to = {address: {index: 1234n, subindex: 0n}, hookName: 'someReceiveHookName'} // A contract receiver can be specified as such.
+const tokenAmount = 100n;
+
+const signer: AccountSigner = ...; // Relies on using private keys of sender account.
+const update = { from, to, tokenAmount, tokenId };
+// const update [{ from, to, tokenAmount, tokenId }, { from, to, tokenAmount, tokenId }] // Example of batch update.
+
+const txHash = await contract.transfer(
+    {
+        senderAddress: from,
+        energy: 10000n,
+    },
+    update,
+    signer
+);
+```
+
+### Using a function for processing the transaction.
+This delegates processing the transaction to a callback function. Typically, this function would submit the transaction through a Concordium compatible wallet, which handles signing and submitting.
+
+```js
+const tokenId = ''; // HEX string representing a token ID defined in the contract.
+const from = '4UC8o4m8AgTxt5VBFMdLwMCwwJQVJwjesNzW7RPXkACynrULmd';
+const to = '3ybJ66spZ2xdWF3avgxQb2meouYa7mpvMWNPmUnczU8FoF8cGB'; // An account receiver.
+// const to = {address: {index: 1234n, subindex: 0n}, hookName: 'someReceiveHookName'} // A contract receiver can be specified as such.
+const tokenAmount = 100n;
+
+const update = { from, to, tokenAmount, tokenId };
+// const update [{ from, to, tokenAmount, tokenId }, { from, to, tokenAmount, tokenId }] // Example of batch update.
+
+const txHash = await contract.transfer(
+    { energy: 10000n },
+    update,
+    ({
+        type, // The transaction type.
+        payload, // The transaction payload
+        parameter: {
+            json, // The parameter to be supplied to the contract receive function in JSON format.
+            hex, // The parameter to be supplied to the contract receive function as a hex encoded string
+        },
+        schema: {
+            value, // The contract schema for the parameter. This is needed to translate the JSON format to a byte array.
+            type, // The type of the schema. This is always 'parameter', meaning it can be used for the JSON formatted parameter only.
+        }
+    }) => {
+        // Do something with the transaction details
```

Comment:
What do you mean, if the account does not exist or if the format is invalid? We do some checks during serialization of lengths according to what is specified in the CIS-2 spec, and use the `AccountAddress` class for account addresses (which has a length check internally), but not much otherwise.

Reply:
I guess my question is, how does the caller handle errors, should they check stuff before calling this, or will there be thrown an exception or does the callback provide an error.

Reply:
It's been changed now (see https://github.com/Concordium/concordium-node-sdk-js/pull/168#discussion_r1182523474)

Documentation has been updated to reflect the change as well.